packages/client/components/TeamPrompt/WorkDrawer/GitHubObjectCard.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react'
+import CopyToClipboard from 'react-copy-to-clipboard'
+import gitHubSVG from '../../../styles/theme/images/graphics/github-circle.svg'
+import gitHubMerged from '../../../styles/theme/images/graphics/github-merged.svg'
+import githubIssueClosed from '../../../styles/theme/images/graphics/github-issue-closed.svg'
+import githubIssueOpen from '../../../styles/theme/images/graphics/github-issue-open.svg'
+import githubPROpen from '../../../styles/theme/images/graphics/github-pr-open.svg'
+import githubPRDraft from '../../../styles/theme/images/graphics/github-pr-draft.svg'
+import githubPRClosed from '../../../styles/theme/images/graphics/github-pr-closed.svg'
+import relativeDate from '../../../utils/date/relativeDate'
+import {Link} from '@mui/icons-material'
+import useTooltip from '../../../hooks/useTooltip'
+import {MenuPosition} from '../../../hooks/useCoords'
+import {useFragment} from 'react-relay'
+import graphql from 'babel-plugin-relay/macro'
+import {GitHubObjectCard_result$key} from '../../../__generated__/GitHubObjectCard_result.graphql'
+
+const ISSUE_STATUS_MAP: Record<string, any> = {
+  OPEN: githubIssueOpen,
+  CLOSED: githubIssueClosed
+}
+
+const PR_STATUS_MAP: Record<string, any> = {
+  OPEN: githubPROpen,
+  DRAFT: githubPRDraft,
+  CLOSED: githubPRClosed,
+  MERGED: gitHubMerged
+}
+
+interface Props {
+  resultRef: GitHubObjectCard_result$key
+}
+
+const GitHubObjectCard = (props: Props) => {
+  const {resultRef} = props
+
+  const result = useFragment(
+    graphql`
+      fragment GitHubObjectCard_result on _xGitHubSearchResultItem {
+        __typename
+        ... on _xGitHubIssue {
```

Comment:
+2 did you get frustrated like me hoping they made `_xGitHubIssueOrPullRequest` an interface instead of a union ? Seems like such a missed opportunity to simplify queries

Reply:
Yessss I poked around the GH GQL API expecting some interface like this but none of the interfaces these implement represented the two of these sufficiently :(